Date selection function implemented

Xi-Zhang1 opened the pull request.

Xi-Zhang1 committed:
Date selection function implemented

Xi-Zhang1 committed:
Refactor Detail screen to re-render after successful data addition

This commit enhances the user experience by ensuring the Detail screen fetches and displays updated data whenever it is revisited after a successful addition of information from the AddPage. This update leverages the useFocusEffect hook from @react-navigation/native to trigger a data refresh every time the Detail screen comes into focus.

wenshilanR committed:
handle user is not logged in

wenshilanR commented:
The display for selecting the date is weird (ios system)

<img width="366" alt="Screenshot 2024-04-29 at 14 40 18" src="https://github.com/JoyfulTechLauncher/joyfulfrugalista/assets/90823176/165b2602-6e99-4339-b55a-27479dc78467">

Xi-Zhang1 committed:
Modify the name of My stack

Xi-Zhang1 committed:
Implement dedicated DateTimePicker rendering for iOS

This commit introduces a modal-based DateTimePicker presentation for iOS devices, ensuring that the date selection interface is immediately accessible upon user interaction.

Xi-Zhang1 committed:
Rollback app.json
